src/middlewared/middlewared/plugins/smb_/registry.py

```diff
@@ -249,18 +246,6 @@ async def add_multiprotocol_conf(self, conf, gl, data):
                                            data['id'], {'cifs_durablehandle': False})
                 data['durablehandle'] = False
 
-        if any(filter(lambda x: f"{x['path']}/" in f"{conf['path']}/" or f"{conf['path']}/" in f"{x['path']}/", gl['afp_shares'])):
```

Comment:
Rather than removing this. We can simply change this check to see whether this is a legacy_afp_share. This will keep logic of only applying parameters to running configuration / not write to DB.